### PR TITLE
Continue creating transport creds when tokenManager is nil.

### PR DESCRIPTION
**Pull request by rmehta19**
Continue creating transport creds when tokenManager is nil.

### CODE REVIEW

File: internal/v2/s2av2.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func NewClientCreds(s2av2Address string, localIdentity *commonpbv1.Identity, ver
 	// Create an AccessTokenManager instance to use to authenticate to S2Av2.
 	accessTokenManager, err := tokenmanager.NewSingleTokenAccessTokenManager()
 	if err != nil {
```
Comment (matthewstevenson88) [inline internal/v2/s2av2.go:64]: S

File: internal/v2/s2av2.go [hunk 2]
```diff
@@ -84,18 +88,22 @@ func NewServerCreds(s2av2Address string, localIdentities []*commonpbv1.Identity,
...
-		return nil, err
+		accessTokenManager = nil
```
Comment (matthewstevenson88) [inline internal/v2/s2av2.go:91]: Same here, should we just delete this if statement?

File: internal/v2/s2av2.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func NewClientCreds(s2av2Address string, localIdentity *commonpbv1.Identity, ver
...
```
[the remark this answers is not in the document]
Comment (rmehta19) [reply internal/v2/s2av2.go:65]: Good point! Done.

File: internal/v2/s2av2.go (hunk 2 above)
Comment (rmehta19) [reply internal/v2/s2av2.go:91]: Good point! Done.

File: internal/v2/fakes2av2/fakes2av2_test.go [hunk 4]
```diff
@@ -200,6 +200,44 @@ func TestSetUpSession(t *testing.T) {
 			},
 			expErr: errors.New("rpc error: code = Unknown desc = SessionReq has no AuthenticationMechanism with a valid token"),
 		},
+		{
+			description: "Get server TLS config -- empty authmechanisms (S2A_ACCESS_TOKEN env var not set",
```
Comment (matthewstevenson88) [inline internal/v2/fakes2av2/fakes2av2_test.go:204]: nit: open parenthesis in the description

File: internal/v2/tls_config_store/tls_config_store_test.go [hunk 5]
```diff
@@ -141,6 +142,17 @@ func TestTLSConfigStoreClient(t *testing.T) {
 	}{
 		{
 			description:        "static",
+			tokenManager:       nil,
```
Comment (matthewstevenson88) [inline internal/v2/tls_config_store/tls_config_store_test.go:145]: Can we add the analogous test case for the GetTLSConfigurationForServer API?

**Review by matthewstevenson88 — APPROVED** [review]
Thanks for the PR Riya! LGTM with 2 nits. :)

File: internal/v2/tls_config_store/tls_config_store_test.go (hunk 5 above)
Comment (rmehta19) [reply internal/v2/tls_config_store/tls_config_store_test.go:145]: Will do!

**Comment by rmehta19** [comment]
Thanks for the review @matthewstevenson88 !

**Pull request merged**